### PR TITLE
simplify and adapt z_zone.c to DSDA-Doom

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Targeted post-release. We need to fix some bugs first before introducing new ones. 😉

File: src/z_zone.c [hunk 1]
```diff
@@ -38,705 +38,170 @@
...
   void **user;
-  unsigned char tag,vm;
-
-#ifdef INSTRUMENTED
-  unsigned short extra;
-  const char *file;
-  int line;
-#endif
-
+  unsigned char tag;
```
Comment (fabiangreffrath) [inline src/z_zone.c:49]: Why not `int`?

File: src/z_zone.c [hunk 2]
```diff
@@ -46,10 +49,10 @@ typedef struct memblock {
   struct memblock *next,*prev;
   size_t size;
   void **user;
-  unsigned char tag;
+  int tag;
```
Comment (rfomin) [inline src/z_zone.c:51]: If we care about alignment, is it 64-bit or 32-bit version? With 64-bit build currently we have:
```
typedef struct memblock {
  unsigned id;                 // 4 bytes
  struct memblock *next,*prev; // 2 * 8 bytes
  size_t size;                 // 8 bytes
  void **user;                 // 8 bytes
  int tag;                     // 4 bytes
} memblock_t;
```
I always forget the exact data alignment/padding rules, usually I just reorder the fields by size. For the 64-bit version, there are a lot of padding the old Doom structs.

Comment (fabiangreffrath) [reply src/z_zone.c:51]: Chunk size is currently 64 byte on a 64-bit system. Maybe that's a bit too much.

Comment (rfomin) [reply src/z_zone.c:51]: Should we care about alignment in structures? If yes, then we need to reorder the `struct memblock`, currently there is padding:
```
typedef struct memblock {
  unsigned id;                 // 4 bytes
  // pad 4 bytes
  struct memblock *next,*prev; // 2 * 8 bytes
  size_t size;                 // 8 bytes
  void **user;                 // 8 bytes
  int tag;                     // 4 bytes
} memblock_t;
```
This is my favorite article on subject: http://www.catb.org/esr/structure-packing/#_structure_alignment_and_padding

File: src/z_zone.h [hunk 3]
```diff
@@ -47,68 +47,26 @@
...
-// Make Z_CheckHeap() more diagnostic
-//
-// Revision 1.3  1998/02/02  13:28:06  killough
-// Add dprintf
-//
-// Revision 1.2  1998/01/26  19:28:04  phares
-// First rev with no ^Ms
-//
-// Revision 1.1.1.1  1998/01/19  14:03:06  rand
-// Lee's Jan 19 sources
```
Comment (rfomin) [inline src/z_zone.h:111]: Small thing, but I like these old logs, even if they are not relevant anymore. What do you think about adding old Boom team logs to repository? https://github.com/Doom-Utils/historic-ports/blob/mbf/log_lee.txt

Comment (fabiangreffrath) [reply src/z_zone.h]: No problem, I have added the historic changelogs back. Feel free to add the Boom team log to our repository, I didn't even know about its existence yet.

**Pull request merged**